iothub/device/src/Transport/AmqpIoT/AmqpIoTReceivingLink.cs

```diff
@@ -272,67 +272,57 @@ private void OnTwinChangesReceived(AmqpMessage amqpMessage)
+                    Logging.Info("Updated twin reported properties successfully", $"{nameof(OnTwinChangesReceived)}");
+                }
+                else if (correlationId.StartsWith(AmqpTwinMessageType.Put.ToString(), StringComparison.OrdinalIgnoreCase))
+                {
+                    // This is an acknowledgement received from service for subscribing to desired property updates
+                    Logging.Info("Subscribed for twin successfully", $"{nameof(OnTwinChangesReceived)}");
                 }
                 else
                 {
-                    Twin twin = null;
-                    TwinCollection twinProperties = null;
-
-                    if (correlationId == null)
-                    {
-                        // Here we are getting desired property update notifications and want to handle it first
-                        using (StreamReader reader = new StreamReader(amqpMessage.BodyStream, System.Text.Encoding.UTF8))
-                        {
-                            string patch = reader.ReadToEnd();
-                            twinProperties = JsonConvert.DeserializeObject<TwinCollection>(patch);
-                        }
-                    }
-                    else if (correlationId.StartsWith(AmqpTwinMessageType.Get.ToString(), StringComparison.OrdinalIgnoreCase))
-                    {
-                        // This a response of a GET TWIN so return (set) the full twin
-                        using (StreamReader reader = new StreamReader(amqpMessage.BodyStream, System.Text.Encoding.UTF8))
-                        {
-                            string body = reader.ReadToEnd();
-                            var properties = JsonConvert.DeserializeObject<TwinProperties>(body);
-                            twin = new Twin(properties);
-                        }
-                    }
-                    else if (correlationId.StartsWith(AmqpTwinMessageType.Patch.ToString(), StringComparison.OrdinalIgnoreCase))
-                    {
-                        // This can be used to coorelate success response with Updating reported properties
-                    }
-                    else if (correlationId.StartsWith(AmqpTwinMessageType.Put.ToString(), StringComparison.OrdinalIgnoreCase))
-                    {
-                        // This is an acknowledgement received from service for subscribing to desired property updates
-                        twin = new Twin();
-                    }
-                    else
-                    {
-                        // This shouldn't happen
-                        // Do nothing
-                    }
-                    _onDesiredPropertyReceived.Invoke(twin, correlationId, twinProperties);
+                    // This shouldn't happen
+                    Logging.Info("Received a correlation Id for Twin operation that does not match Get, Patch or Put request", $"{nameof(OnTwinChangesReceived)}");
```

Comment:
FWIW, the `$"{` and `}"` around `nameof(OnTinChangesReceived)` is not needed. It is a string and you aren't formatting anything else into the string, so it is pointless work.